stock_account_internal_move/models/stock_move.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (C) 2018 by Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, models
+from odoo.tools import float_round
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    # @override
+    @api.multi
+    def _action_done(self):
+        """Call _account_entry_move for internal moves as well."""
+        res = super()._action_done()
+        for move in res:
+            # first of all, define if we need to even valuate something
+            if move.product_id.valuation != 'real_time':
+                continue
+            # we're customizing behavior on moves between internal locations
+            # only, thus ensuring that we don't clash w/ account moves
+            # created in `stock_account`
+            if not move._is_internal():
+                continue
+            move._account_entry_move()
+        return res
+
+    # @override
+    @api.multi
+    def _run_valuation(self, quantity=None):
+        # Extend `_run_valuation` to make it work on internal moves.
+        self.ensure_one()
+        res = super()._run_valuation(quantity)
+        if self._is_internal() and not self.value:
+            # TODO: recheck if this part respects product valuation method
```

Comment:
were the todo checked?

Reply:
Hi, sorry, I am not sure what is the intention of the original author too.
Any hint, @max3903?